### PR TITLE
feat(spacing): add spacing options for Grid and Layout

### PR DESCRIPTION
**Pull request by arahansen**
- adds margin{Top, Right, Bottom, Left} and padding{Top, Right, Bottom, Left} properties to `Grid` and `Layout` (margin only on Layout).
- If margin props and margin value props are included, an error will be thrown in development.
- This PR should also update some examples to use this new syntax.

### CODE REVIEW

**Comment by obartra** [comment]
Did the bundle size increase?

File: src/grid.js [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
   padding?: Spacing,
   size?: Size,
+  marginTop?: SpacingValues,
+  marginRight?: SpacingValues,
+  marginBottom?: SpacingValues,
+  marginLeft?: SpacingValues,
+  paddingTop?: SpacingValues,
+  paddingRight?: SpacingValues,
+  paddingBottom?: SpacingValues,
+  paddingLeft?: SpacingValues,
```
Comment (obartra) [inline src/grid.js]: this is reeeally nit picky but you could sort them alphabetically with something like https://github.com/Tyriar/vscode-sort-lines

File: src/layout.js [hunk 2]
```diff
@@ -61,24 +61,26 @@ export default class Layout extends React.Component {
     margin?: Spacing,
     overflow?: Overflow,
     type?: LayoutType,
+    children: React$Element<*>,
```
Comment (obartra) [inline src/layout.js]: Why is children needed? It looks like it's treated like any other property

File: src/utils.js [hunk 3]
```diff
@@ -30,6 +33,7 @@ export function getDisplayName(WrappedComponent: string | Component): string {
 function getSizeName(size: number) {
   switch (size) {
     case 0:
+    case undefined:
```
Comment (obartra) [inline src/utils.js]: Undefined is "single" in terms of class names, should not be 0 unless you update the css

File: src/utils.js [hunk 4]
```diff
@@ -77,4 +112,7 @@ function getSpacing(
   )
 }
 
-export const getSpacingClasses = memoize(getSpacing, stringify)
+export const getSpacingClasses = memoize(
+  getSpacing,
+  (props: GridProps, type: string) => type + stringify(props)
```
Comment (obartra) [inline src/utils.js]: This is fine but I expected `eslint` would yell at you for not interpolating

**Comment by arahansen** [comment]
closing in favor of https://github.com/obartra/reflex/pull/189

**Pull request closed without merging**